src/esp/assets/GenericMeshData.h

```diff
@@ -52,23 +52,35 @@ class GenericMeshData : public BaseMesh {
    */
   virtual void uploadBuffersToGPU(bool forceReload = false) override;
 
+  /**
+   * @brief Set mesh data from external source, and sets the @ref collisionMesh_
+   * references.  Can be used for meshDatas that are manually synthesized, such
+   * as NavMesh. Sets the @ref collisionMesh_ references.
+   * @param meshData the meshData to be assigned.
+   */
+  void setMeshData(
+      Corrade::Containers::Optional<Magnum::Trade::MeshData>& meshData);
+
   /**
    * @brief Load mesh data from a pre-parsed importer for a specific mesh
-   * component. Sets the @ref collisionMeshData_ references.
-   * @param importer The importer pre-loaded with asset data from file.
+   * component ID. Sets the @ref collisionMeshData_ references.
+   * @param importer The importer pre-loaded with asset data from file, or a
+   * Primitive Importer.
    * @param meshID The local identifier of a specific mesh component of the
    * asset.
    */
-  void setMeshData(Magnum::Trade::AbstractImporter& importer, int meshID);
+  void importAndSetMeshData(Magnum::Trade::AbstractImporter& importer,
```

Comment:
```suggestion
  void setImportedMeshData(Magnum::Trade::AbstractImporter& importer,
```

Maybe? The "and set" seems superfluous to me. Same below. (It's a matter of personal preference I guess, feel free to ignore.)

Reply:
Wished to specify that "setMeshData" was also called.